src/Compilers/Core/Desktop/CommandLine/CommonCompiler.cs

```diff
@@ -258,17 +254,31 @@ internal bool PrintErrors(IEnumerable<DiagnosticInfo> diagnostics, TextWriter co
             return hasErrors;
         }
 
-        internal virtual void PrintError(DiagnosticInfo diagnostic, TextWriter consoleOutput)
+        protected virtual void PrintError(DiagnosticInfo diagnostic, TextWriter consoleOutput)
         {
             consoleOutput.WriteLine(diagnostic.ToString(Culture));
         }
 
+        private ErrorLogger GetErrorLogger(TextWriter consoleOutput, CancellationToken cancellationToken)
+        {
+            Debug.Assert(Arguments.ErrorLogPath != null);
+
+            var errorLog = OpenFile(Arguments.ErrorLogPath, consoleOutput, FileMode.Create, FileAccess.Write, FileShare.ReadWrite | FileShare.Delete);
+            if (errorLog == null)
+            {
+                return null;
+            }
+
+            return new ErrorLogger(errorLog, GetToolName(), GetAssemblyFileVersion(), GetAssemblyVersion());
+        }
+
         /// <summary>
         /// csc.exe and vbc.exe entry point.
         /// </summary>
-        public virtual int Run(TextWriter consoleOutput, CancellationToken cancellationToken)
+        public virtual int Run(TextWriter consoleOutput, CancellationToken cancellationToken = default(CancellationToken))
         {
             var saveUICulture = Thread.CurrentThread.CurrentUICulture;
```

Comment:
Probably the core command line compiler is always invoked single threaded, though compiler team experts in this area would know better than me.